src/rade_api.h

```diff
@@ -0,0 +1,95 @@
+/*---------------------------------------------------------------------------*\
+
+  rade_api.h
+
+  Library of API functions that implement the Radio Autoencoder API.
+
+\*---------------------------------------------------------------------------*/
+
+/*
+  Copyright (C) 2024 David Rowe
+
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions
+  are met:
+
+  - Redistributions of source code must retain the above copyright
+  notice, this list of conditions and the following disclaimer.
+
+  - Redistributions in binary form must reproduce the above copyright
+  notice, this list of conditions and the following disclaimer in the
+  documentation and/or other materials provided with the distribution.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+  ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+  A PARTICULAR PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE FOUNDATION OR
+  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+#ifndef __RADE_API__
+#define __RADE_API__
+
+#include <sys/types.h>
+
+// This declares a single-precision (float) complex number
+
+#ifndef __RADE_COMP__
+#define __RADE_COMP__
+typedef struct {
+  float real;
+  float imag;
+} RADE_COMP;
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Sample rates used
+#define RADE_FS_8000 8000           // modem waveform sample rate
+#define RADE_FS_16000 16000         // speech sample rate
```

Comment:
Should the names of these #defines be tweaked to match their intended use? For example:

```suggestion
#define RADE_MODEM_SAMPLE_RATE 8000           // modem waveform sample rate
#define RADE_SPEECH_SAMPLE_RATE 16000         // speech sample rate
```

Reply:
Also, typical rule of thumb for preprocessor definitions is to surround them in parenthesis, i.e. `#define XYZ (123)`. Though I don't think that'll matter much here.

Reply:
Why? I've seen parens used when it's an expression, but not a constant.